disperser/dataapi/nonsigner_utils_test.go

```diff
@@ -389,3 +380,96 @@ func TestCreateOperatorQuorumIntervals(t *testing.T) {
 	assert.ElementsMatch(t, []uint8{0x00, 0x02}, quorumIntervals.GetQuorums("operator-2", 22))
 	assert.ElementsMatch(t, []uint8{0x00, 0x02}, quorumIntervals.GetQuorums("operator-2", 25))
 }
+
+func TestComputeNumBatches(t *testing.T) {
+	intervals := []*dataapi.NumBatchesAtBlock{}
+	assert.Equal(t, 0, dataapi.ComputeNumBatches(intervals, 1, 4))
+
+	intervals = []*dataapi.NumBatchesAtBlock{
+		{
+			BlockNumber: 5,
+			AccuBatches: 2,
+		},
+	}
+	assert.Equal(t, 0, dataapi.ComputeNumBatches(intervals, 1, 4))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 1, 5))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 5, 5))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 5, 6))
+
+	intervals = []*dataapi.NumBatchesAtBlock{
+		{
+			BlockNumber: 5,
+			AccuBatches: 2,
+		},
+		{
+			BlockNumber: 10,
+			AccuBatches: 4,
+		},
+		{
+			BlockNumber: 15,
+			AccuBatches: 6,
+		},
+		{
+			BlockNumber: 20,
+			AccuBatches: 8,
+		},
+	}
+
+	assert.Equal(t, 0, dataapi.ComputeNumBatches(intervals, 1, 4))
+	assert.Equal(t, 0, dataapi.ComputeNumBatches(intervals, 21, 22))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 1, 5))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 5, 5))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 5, 9))
+	assert.Equal(t, 4, dataapi.ComputeNumBatches(intervals, 5, 10))
+	assert.Equal(t, 2, dataapi.ComputeNumBatches(intervals, 6, 10))
```

Comment:
so `ComputeNumBatches(intervals, 6, 10))` returns 2 because that's the number of batches produced between 6 and 10. 
Then should `ComputeNumBatches(intervals, 5, 10))` be also 2? The first interval doesn't have the `NumBatches` field which implies that the 2 batches were produced earlier than block 5. Am i misunderstanding this?